eventmesh-protocol-plugin/eventmesh-protocol-meshmessage/src/main/java/org/apache/eventmesh/protocol/meshmessage/resolver/tcp/TcpMessageProtocolResolver.java

```diff
@@ -77,7 +75,11 @@ public static CloudEvent buildEvent(Header header, EventMeshMessage message) thr
             .withSource(URI.create("/"))
             .withType("eventmeshmessage")
             .withSubject(topic)
-            .withData(content.getBytes(StandardCharsets.UTF_8));
+            .withData(content.getBytes(Constants.DEFAULT_CHARSET));
+
+        if (message.getHeaders().containsKey(Constants.DATA_CONTENT_TYPE)) {
+            cloudEventBuilder.withDataContentType(message.getHeaders().get(Constants.DATA_CONTENT_TYPE));
+        }
 
```

Comment:
Since you implement by adding headers, you also need to set headers in TcpMessageProtocolResolver#buildEventMeshMessage.

```
if (cloudEvent.getDataContentType() != null) {
           eventMeshMessage.getHeaders().put(Constants.DATA_CONTENT_TYPE, cloudEvent.getDataContentType());
}
```

Reply:
resolved. thanks for notification!